### ISSUE TITLE
 battery charge

### ISSUE THREAD

**Issue by Mert0035** [lery600/obstacleAvoidingRobot#1]
How did you handle charging of 18650 batteries? If we connect an adapter on bms, will the batteries be charged?

**Comment by lery600** [comment]
the voltage you are connecting to pins for charge and discharge should be 12.6v. then it just charges. of course  more convenient would be a charger for lithium batteries. but i don't have one. if you are working with 18650 cells it's important to fully charge them so they have the same voltage. all of them. it must be done before connecting them together.

**Comment by Mert0035** [comment]
I have 4 lithium ion batteries. do you think it would be enough how many mah batteries you use? I have 2500 mah 4 batteries. Did you use a brushless motor to have a strong vacuum? If we use a dc motor, it might be insufficient, right?

**Comment by lery600** [comment]
i have one motor from a car vacuum cleaner. it's a brushed dc motor and it looks like it's actually better than brushless motor i use in that robot (i use the cheapest brushless motor). so i would say a good brushed motor is better because it's easier to control.

**Comment by lery600** [comment]
the only option i see here is to put 3 batteries for achieving 12 volts. but as i can see you don't use modules which demand this voltage so you could use all 4 of them and achieve 5000mah capacity. in my case i use 3000mah batteries. so i have 9000mah 12v power source in general. brushless motors are too power demanding and too expensive to control.

**Comment by Mert0035** [comment]
Yes for now I have 11.1 volts 5000 mA with 4 batteries. I connected them to 4s 40A circuit, they are charging. I use Starting materials, except for the brushless motor. I think a simple DC motor will do the job. Have some. at least I don't need the driver, I can run it by connecting it directly to the switch.

**Comment by Mert0035** [comment]
I also did not use a microswitch. Do you think it creates a problem in his work? because I position the materials on plexiglass as a chassis. Unfortunately i don't have 3d printera.

**Comment by lery600** [comment]
good. i think what you make it of is just a matter of convenience. if you do a lot of custom things 3d printer it helpful a lot. but if it's a one time build better use whatever you have. i used acrylic or aluminum profiles from a hardware store before 3d printer. also the material which it was printed of is kind of disappointing. i shouldn't design parts which rely on flexibility with this material. as i said it needs a big revision.

**Comment by Mert0035** [comment]
I tried running the circuit without brushless motor, esc and lion batteries but it didn't work. at least I'd try to get the snake move done. I gave 5v to the drivers with arduino uno. Did I make a mistake somewhere? As far as I can see from the circuit, the li-on batteries only feed the brushless motor, I don't think it has anything to do with it.
https://drive.google.com/file/d/1w47IbC5ZIl0MJ34mP9kYcc5i1-UqBCqr/view?usp=sharing

**Comment by Mert0035** [comment]
In addition, although I gave 12v to the vmotor input, there was still no movement.

**Comment by Mert0035** [comment]
Could there be any missing part in the circuit diagram in your youtube video? Or it may be overlooked because the image is not very clear.

**Comment by lery600** [comment]
if you connected ground and 12v from your battery to a motor and it doesn't spin it means your motor is broken. 
was there a smoke? does this motor need exactly 12 volts?

**Comment by lery600** [comment]
also sometimes BMS can not work properly. if you didn't check the voltage on BMS output it's better to check before blaming the motor.

**Comment by Mert0035** [comment]
I tried the engine with a 12v adapter and it works. I energized the circuit with the adapter other than bms, but still there was no movement. Is there a clearer version of this circuit diagram? The video is not exactly clear, maybe there may be overlooked places.

**Comment by Mert0035** [comment]
I installed it again. there is a moaning noise in the engines, but no rotation. I also changed the voltage setting from the pot on the drv8825 but it still did not work. How many volts did you set the pot?

**Comment by Mert0035** [comment]
your li-on batteries directly power the brushless motor. The energy of other step motors is not visible in the diagram. I've tried all sorts of shapes but have failed to get it working can you help me? I'm running out of time for me to turn in the homework.

**Comment by lery600** [comment]
![the whole picture prepared 0002](https://user-images.githubusercontent.com/63790737/100823785-ec8df680-3465-11eb-9bdd-9376fb204b83.png)

**Comment by lery600** [comment]
i pointed driver connections for battery with thick red here^. before it was just called VMOT. i just didn't expect it would be someone's first project. quite a lot of stuff to learn for the first time.
can you attach a photo? would be much easier to guess what the problem is.

**Comment by lery600** [comment]
btw it's normal when nothing is working. could be pretty disappointing sometimes. very often turns out that i put a small wire in a wrong place  and the whole thing doesn't work.

**Comment by lery600** [comment]
if you use only brushed motors don't look at those motors connections. they are controlled by complicated drivers. it's another whole topic. if you are making a simple brushed version. for which you need other drivers. for example arduino motor shield, or sparkfun monster motor shield or just a simple driver for brushed DC motor

**Comment by lery600** [comment]
![SharedScreenshot](https://user-images.githubusercontent.com/63790737/100821047-a1251980-3460-11eb-8f92-04e1a810f0df.jpg)

**Comment by lery600** [comment]
i have this one for example^

**Comment by lery600** [comment]
![20201202_053253](https://user-images.githubusercontent.com/63790737/100821090-b7cb7080-3460-11eb-8235-6e50c31c812a.jpg)

**Comment by lery600** [comment]
to eliminate all potential misunderstandings i quickly built the circuit you should have with brushed motors. on this driver there are motor A and motor B connectors for your motors. on the other side of this driver board you connect your arduino pwm-capable pins to "B-1A" and "B-1B" connectors for B motor. then "A-1A" and "A-1B" for A motor. the power should be connected directly to the driver (black and white disconnected wires on the picture) and then the arduino should be connected to your power source with VCC (or RAW in some versions) and ground(green and yellow disconnected wires on the picture). so that means this white wire and yellow should be connected together and to the "+" of your battery. and the black and green wires should be connected together and to the "-" of your battery. btw all that is on the picture will work even with 5volts. 
IMPORTANT - VCC is not 5v. don't connect your battery to 5v of arduino. 
may be i said something that you already know but you are on a hurry so i compensate for the time i'm not available))

**Comment by lery600** [comment]
and do not connect motors directly to arduino or it will burn. as for the turbine motor you should just connect it directly to the power source and one of the wires should be disconnected by a switch, no connection to arduino needed
.

**Comment by lery600** [comment]
" I also changed the voltage setting from the pot on the drv8825 but it still did not work." do you have steppers? are they NEMA17?
for example once i pulled a stepper out of a broken printer that looks like NEMA17 but apparently it's very different. with DRV8825 it just made noise and didn't move.

**Comment by lery600** [comment]
in general i have too little idea of your situation. it's better to send photos.

**Comment by Mert0035** [comment]
I was actually planning to set it up with the small dc motor in the photo you took. but since I have little coding knowledge, I chose to do the same with yours. The most important issue for me was the zigzag movement like a tractor plowing the field. you only have all the details.

**Comment by Mert0035** [comment]
![WhatsApp Image 2020-12-02 at 12 04 59](https://user-images.githubusercontent.com/74210192/100851996-f5e38700-3496-11eb-9cce-8d928815e3b4.jpeg)
![WhatsApp Image 2020-12-02 at 12 04 58 (1)](https://user-images.githubusercontent.com/74210192/100852003-f714b400-3496-11eb-81ad-fa11009d37eb.jpeg)
![WhatsApp Image 2020-12-02 at 12 04 58](https://user-images.githubusercontent.com/74210192/100852005-f714b400-3496-11eb-83d9-c1b5c99a9d77.jpeg)
![WhatsApp Image 2020-12-02 at 12 04 59 (1)](https://user-images.githubusercontent.com/74210192/100852010-f7ad4a80-3496-11eb-83aa-0c22f960e330.jpeg)

**Comment by Mert0035** [comment]
in general my circuit is like this. If I do a zigzag movement, that's enough for me now. I can also vacuum with a computer fan or dc motor.

**Comment by Mert0035** [comment]
![WhatsApp Image 2020-12-02 at 12 14 34](https://user-images.githubusercontent.com/74210192/100853221-92f2ef80-3498-11eb-8b64-194e2766b676.jpeg)

**Comment by Mert0035** [comment]
If I can't do the same as yours I'll take this wheel and try to make a robot . what do you think i should do Can it work if I try with a different driver? "for example once i pulled a stepper out of a broken printer that looks like NEMA17 but apparently it's very different. with DRV8825 it just made noise and didn't move." Have you been able to start the engine here later? where is the problem.

**Comment by Mert0035** [comment]
If I use dc instead of step, it has a coding. Did you learn how to code at school or did you yourself? they taught us very little at school. it is also insufficient for such a project.

**Comment by Mert0035** [comment]
![pixiz-02-12-2020-11_02_29](https://user-images.githubusercontent.com/74210192/100858285-d7818980-349e-11eb-89ea-63118d49efb3.jpg)

**Comment by Mert0035** [comment]
I tried it with my li-on batteries and 12v adapter but it didn't work. maybe the adapter's current is not enough. Do you think my notebook adapter is too high?

**Comment by lery600** [comment]
ok than before doing anything with the code take resistance measurment of wires of the motor. this way you will understand which ones are parts of one coil. they should act as two coils with two wires each. then if you know where are the coils - connect the motor as it shown on the picture below.
as i assume your steppers have 72 steps per rotation. nema17 has 200. you can play with this value #define MICROSTEPS 16
and this value int speeddelay = 10; just change the number.

**Comment by lery600** [comment]
![driver](https://user-images.githubusercontent.com/63790737/100898103-7626de00-34d1-11eb-9b42-3a9b4b5e4668.png)

**Comment by lery600** [comment]
your notebook adapter has too high voltage, and the phone adapter has too little current.

**Comment by lery600** [comment]
with adapters you mean just one motor doesn't work? your battery shows the voltage on a multimiter but it doesn't turn anything on? btw with 4 cells you have to have 16.8 volts. if you have lower voltage it's better to charge it.

**Comment by Mert0035** [comment]
yes you are right, the current is too low in the phone adapter. I'll try again but I don't have much hope frankly. Because I've tried so many times for days, but all of them failed.

**Comment by Mert0035** [comment]
When I turn it on with the phone adapter, it only produces a thin sound like moaning. there is no rotational movement in the motors.  I wish I had more time, but on the other hand, I cannot spare much time since I have other classes and exams.

**Comment by lery600** [comment]
i added a version for brushed motors, look - https://github.com/lery600/obstacleAvoidingRobot/blob/master/obstacle%20avoiding%20with%20brushed%20(simplified)

**Comment by lery600** [comment]
as for battery you can even use a usual crona 9v battery it just won't work for a long time.

**Comment by Mert0035** [comment]
thank you so much. I'm going to go get the little robot wheelset today and upload the codes. I hope I will succeed this time.

**Comment by Mert0035** [comment]
I'm asking you a lot of questions and bothering, I'm sorry. I built the circuit, but the wheels go back and forth. I could not find the same driver you are using. Although I have looked at many places. I used the L293d integration.

**Comment by Mert0035** [comment]
![indir](https://user-images.githubusercontent.com/74210192/101260536-cf954400-3741-11eb-8ca0-33d1ef4c5169.jpg)

**Comment by Mert0035** [comment]
[WhatsApp Video 2020-12-05 at 21.53.17.zip](https://github.com/lery600/obstacleAvoidingRobot/files/5647728/WhatsApp.Video.2020-12-05.at.21.53.17.zip)

**Comment by Mert0035** [comment]
I have been working for 2 days, but the engines are constantly turning. the sensor has no effect.

**Comment by Mert0035** [comment]
![WhatsApp Image 2020-12-05 at 21 55 10](https://user-images.githubusercontent.com/74210192/101261385-e63e9980-3747-11eb-9c8e-9396d0a53168.jpeg)

**Comment by Mert0035** [comment]
actually I bought it from this shield but I couldn't use it. It takes up all the inputs on the arduino. Connecting the ultrasonic sensor would be a problem. I removed the integrated on it and used it.

**Comment by Mert0035** [comment]
Actually, it's not a very difficult thing. There is a constant mishap, I don't understand, it's gone.:))

**Comment by Mert0035** [comment]
![l293d-bağlantı-şeması](https://user-images.githubusercontent.com/74210192/101261873-39661b80-374b-11eb-8c9f-3040a40e27e3.png)

**Comment by Mert0035** [comment]
I connected as in the photo. I tried many different connection types but in vain. I tried the motor connection and the sensor connection over and over but nothing changed.

**Comment by Mert0035** [comment]
It happens even though it is no longer promini. The uno model is fine, but editing the code didn't help. Maybe I couldn't do it. but when I did it for the first time I connected the ultrasonic sensor to the pro mini as shown in the video. but the sensor had no effect, the wheels just turned and nothing else happened.
